### PR TITLE
dtv: Remove manual memory management

### PR DESCRIPTION
**Pull request by ThomasHabets**
Not tested end to end.

### CODE REVIEW

**Comment by drmpeg** [comment]
```make test``` does test this block.

**Comment by ThomasHabets** [comment]
Sure, I'm just making it clear that for this one I've not booted up a running grc and make sure that it still works end to end.

**Review by drmpeg — APPROVED** [review]
Tested with live transmission. Works good!

File: gr-dtv/lib/dvb/dvb_ldpc_bb_impl.h [hunk 1]
```diff
@@ -35,7 +35,8 @@ class dvb_ldpc_bb_impl : public dvb_ldpc_bb
     int ldpc_lut_index[FRAME_SIZE_NORMAL];
     void ldpc_lookup_generate(void);
 
-    uint16_t** ldpc_lut;
+    std::vector<uint16_t*> ldpc_lut; // Pointers into ldpc_lut_data.
```
Comment (mbr0wn) [inline gr-dtv/lib/dvb/dvb_ldpc_bb_impl.h:38]: Thanks for adding that comment! It's really helpful in this context. It avoids the "uh why do we still have raw pointers" question.

**Review by mbr0wn — APPROVED** [review]

**Comment by mormj** [comment]
QA failure appears unrelated.  Merging.

**Pull request merged**